src/views/open-api-docs-view/components/property-details/index.tsx

```diff
@@ -110,21 +129,71 @@ function PropertyDetailsNested({
 						<Badge text="Required" color="highlight" size="small" />
 					) : null}
 				</div>
-				{description ? (
+			</TreeContent>
+			{/* Optional description and nested properties */}
+			{description || hasNestedProperties ? (
+				<TreeContent hideBorder={isLastItem}>
+					{description ? (
+						<TreeContent
+							listItemStyle={!hasNestedProperties ? 'last' : null}
+							hideIndicator={!hasNestedProperties}
+						>
+							<div
+								className={s.nestedDescription}
+								dangerouslySetInnerHTML={{ __html: description }}
+							/>
+						</TreeContent>
+					) : null}
+					{hasNestedProperties ? (
+						<ListNestedProperties
+							nestedProperties={nestedProperties}
+							depth={depth}
+						/>
+					) : null}
+				</TreeContent>
+			) : null}
+		</div>
+	)
+}
+
+/**
+ * Display `children` with structured tree styles, including tree markers.
+ *
+ * Accepts optional props that affect the appearance of the tree styles:
+ * - `hideBorder` - hide the tree marker, but retain spacing
+ * - `hideIndicator` - hide the tree marker, which shifts positioning,
+ *    but still retains a smaller amount of consistent left padding.
+ * - `listItemStyle` - affects the tree marker style. By default,
+ *    the marker is a vertical line that extends to the bottom of the
+ * 	  container, creating the appearance of a continuous line for
+ * 		consecutive list items. The `middle` style adds a horizontal line,
+ *    which points to the children. The `last` style curves the vertical line
+ *    and ends it by pointing to the children.
+ */
+function TreeContent({
```

Comment:
The bulk of the changes centre around this new `TreeContent` component. This component builds some `tree`-like styling around the provided `children`.

Reply:
So nice ✨ 🌳